src/OrchardCore.Modules/OrchardCore.OpenId/OpenIdExtensions.cs

```diff
@@ -22,6 +22,12 @@ internal static string GetUserName(this ClaimsPrincipal principal)
            principal.FindFirst(ClaimTypes.Name)?.Value ??
            throw new InvalidOperationException("No suitable user name can be found in the principal.");
 
+    internal static string[] GetRoles(this ClaimsPrincipal principal)
+        => principal.FindAll(c => c.Type is Claims.Role || c.Type is ClaimTypes.Role)
```

Comment:
Note: with pattern matching, you can simplify that:

```suggestion
        => principal.FindAll(c => c.Type is Claims.Role or ClaimTypes.Role)
```